kpi/utils/monkey_patching.py

```diff
@@ -23,4 +27,42 @@ def create_permissions(app_config, using=DEFAULT_DB_ALIAS, **kwargs):
     return django_create_permissions(app_config=app_config, using=using, **kwargs)
 
 
+def get_candidate_relations_to_delete(opts):
+    """
+    Filter relations that are not from the same database
+
+    The db connection is set at db router level. See `db_for_read()` and
+    `db_for_write()` in `kpi/db_routers.py::DefaultDatabaseRouter` class.
+    Unfortunately, it does not persist for candidates below.
+    So, even if db connection is set to one database, Django could detect
+    candidates to delete, based on `on_delete` attribute, from the other
+    database - which obviously raises an error because the table does not exist.
+    """
+
+    db_connection = router.db_for_write(opts.model)
+
+    return (
+        f
+        for f in opts.get_fields(include_hidden=True)
+        if f.auto_created
+        and not f.concrete
+        and (f.one_to_one or f.one_to_many)
```

Comment:
Can you add a comment specifying that here and below is the actual new stuff, and the bit above comes directly from the original source code?